docs/proposals/scheduling/20230801-descheduler-remove-anomaly-pod.md

```diff
@@ -0,0 +1,210 @@
+| ----------------------- | ------------------------- | ------------------------------------------------------------------------------------------- |
+| minPodStartupSeconds    | uint                      | Minimum startup window time, default 5 minutes                                                               |
+| excludeOwnerKinds       | list(string)              | Exclude some special workload types, such as StatefulSet                                            |
+| eventReasons            | list(string)              | Event Reason with obvious characteristics, such as FailedCreatePodSandbox                                    |
+| reasons                 | list(string)              | Some special message characteristics                                                                     |
+| annotationHints         | list(string)              | A key with an annotation that can give some error messages |
+| includingInitContainers | bool                      | Whether to include initContainer                          |
+| namespaces              | (see [namespace filtering](https://github.com/kubernetes-sigs/descheduler#namespace-filtering)) | The namespace to be processed or not to be processed                                                          |
+| labelSelector           | (see [label filtering](https://github.com/kubernetes-sigs/descheduler#label-filtering))     | Which Pods to process, if not set, it will default to all Pods                             |
+| anomalyCondition        | object                    | Conditions for abnormal judgment                                                                                |
+| nodeDowngradePolicy     | object                    | Node downgrade policies                                                                                |
+
+- `anomalyCondition` is defined as follows:
+By default, if the problem is identified 5 times in a row, it is considered abnormal
+
+```go
+type AnomalyCondition struct {
+	// Timeout indicates the expiration time of the abnormal state, the default is 1 minute
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+	// ConsecutiveAbnormalities indicates the number of consecutive abnormalities
+	ConsecutiveAbnormalities uint32 `json:"consecutiveAbnormalities,omitempty"`
+	// ConsecutiveNormalities indicates the number of consecutive normalities
+	ConsecutiveNormalities uint32 `json:"consecutiveNormalities,omitempty"`
+}
+```
+
+- The `nodeDowngradePolicy` is defined as follows:
+    - The default weight reduction method is label
+    - Default ttl is 5 minutes
+
+```go
+type NodeDowngradeMethod string
+
+const (
+	NodeDowngradeMethodTaint NodeDowngradeMethod = "taint"
+	NodeDowngradeMethodLabel NodeDowngradeMethod = "label"
+)
+
+type NodeDowngradePolicy struct {
+	Method NodeDowngradeMethod `json:"method,omitempty"`
+	TTL    *metav1.Duration    `json:"ttl,omitempty"`
+}
+```
```

Comment:
Lack of security configuration for Node downgrades, such as QPS or the maximum allowable downgrade node ratio.